system/Config/Factories.php

```diff
@@ -74,9 +77,10 @@ class Factories
      *
      * @template T of Model
      *
-     * @param class-string<T> $name
+     * @phpstan-param class-string<T> $name
      *
-     * @return T
+     * @return object
```

Comment:
Technically this doesn't need to return an instance of `CodeIgniter\Model` unless `instanceOf` is specified (which I think `model()` does) but for consistency maybe we should use that for the vanilla return type anyways?
```suggestion
     * @return Model
```